Sources/NIOCore/ByteBuffer-hexdump.swift

```diff
@@ -85,6 +95,46 @@ extension ByteBuffer {
         let endHex = back.hexDumpPlain()
         return startHex + " ... " + endHex
     }
+    
+    /// Return a `String` of  hexadecimal digits of the readable bytes in the buffer,
+    /// analog to `.plain` format but without whitespaces.
+    /// `hexDumpCompact()` always dumps all readable bytes, i.e. from `readerIndex` to `writerIndex`,
+    /// so you should set those indices to desired location to get the offset and length that you need to dump.
+    private func hexDumpCompact() -> String {
+        var hexString = ""
+        hexString.reserveCapacity(self.readableBytes * 3)
+
+        for byte in self.readableBytesView {
+            hexString += String(byte, radix: 16, padding: 2)
+        }
+
+        return hexString
+    }
+
+    /// Return a `String` of  hexadecimal digits of the readable bytes in the buffer,
+    /// analog to `.plain` format but without whitespaces and clips the output to the max length of `maxBytes` bytes.
+    /// If the dump contains more than the `maxBytes` bytes, this function will return the first `maxBytes/2`
+    /// and the last `maxBytes/2` of that, replacing the rest with `...`, i.e. `01 02 03 ... 09 11 12`.
+    ///
+    /// - parameters:
+    ///     - maxBytes: The maximum amount of bytes presented in the dump.
+    private func hexDumpCompact(maxBytes: Int) -> String {
+        // If the buffer length fits in the max bytes limit in the hex dump, just dump the whole thing.
+        if self.readableBytes <= maxBytes {
+            return self.hexDump(format: .plain)
+        }
+
+        var buffer = self
+
+        // Safe to force-unwrap because we just checked readableBytes is > maxBytes above.
+        let front = buffer.readSlice(length: maxBytes / 2)!
+        buffer.moveReaderIndex(to: buffer.writerIndex - maxBytes / 2)
+        let back = buffer.readSlice(length: buffer.readableBytes)!
+
+        let startHex = front.hexDumpCompact()
+        let endHex = back.hexDumpCompact()
+        return startHex + " ... " + endHex
+    }
```

Comment:
Both these methods are almost identical to the `plain` ones. Can we refactor them to reuse the code instead?

Reply:
Ofcourse! Should we factor out common code to a separate function and make the `hexDumpPlain` and `hexDumpCompact` call that function?